tools/generator/Java.Interop.Tools.Generator.Importers/XmlApiImporter.cs

```diff
@@ -295,6 +295,7 @@ static string EnsureValidIdentifer (string name)
 				return name;
 
 			name = name.Replace ('$', '_');
+			name = name.Replace ('-', '_');
```

Comment:
Are we sure that these are the only two characters that need escaping?

Meaning, should we instead use some form of regex to replace everything that isn't a valid C# character?

Related:

  * `ResourceIdentifier.CreateValidIdentifier()` in https://github.com/xamarin/xamarin-android/blob/master/src/Xamarin.Android.Build.Tasks/Utilities/ResourceIdentifier.cs
  * https://github.com/xamarin/xamarin-android/commit/af61ecda302d69dd714f3a363829c3a9e593f5ab

Reply:
Please update to use `IdentifierValidator.CreateValidIdentifier()` from PR #497.